### PR TITLE
Add a release procedure

### LINKED ISSUE #46: Document steps to release dask-jobqueue
I have just made the 0.1.1 release of dask-jobqueue. These are the steps I took: 

- checkout dask/dask-jobqueeue master, for me this was: `git checkout upstream/master`
- create a tag: `git tag -a 0.1.1 -m '0.1.1'`
- push the tag to github: `git push --tags upstream`
- build the wheels/dist for pypi: `python setup.py bdist_wheel sdist`
- upload the release to pypi: `twine upload dist/dask*0.1.1*`

@jakirkham has suggested wrapping these tasks into a `rever` workflow: https://github.com/dask/dask-jobqueue/issues/36#issuecomment-382606275. I'm open to going that route but it is not a high priority for me at the moment.

Perhaps someone else wants to step in there. There may be other steps we want in this workflow. I view this as the bare minimum. I think this issue is the place where we should discuss those.

### PR DESCRIPTION
**Pull request by guillaumeeb**
Closes #46

### CODE REVIEW

File: docs/release-procedure.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+Dask-jobqueue heavily relies on dask and distributed upstream projects.  
+We may want to check their status while releasing.
+
+
+Release for dask-jobqueue, from within your fork:
+
+* Update release notes in `docs/source/changelog.rst`. Preferably within a PR,
+which informs of incoming release.
```
Comment (mrocklin) [inline docs/release-procedure.md]: The last sentence here isn't a full sentence.  I recommend merging them.

Submit a PR that updates the release notes in `docs/source/changelog.rst`.  We submit a PR to inform other developers of the pending release.

File: docs/release-procedure.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+* Once PR is merge, checkout master branch:
```
Comment (mrocklin) [inline docs/release-procedure.md]: Once the PR is merged, checkout the master branch

File: docs/release-procedure.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+git tag -a x.x.x -m 'Version x.x.x'
+git push --tags upstream
+````
+
+* Build the wheel/dist and upload to PyPI:
+
+````
+git clean -xfd
+python setup.py sdist bdist_wheel --universal
+wine upload dist/*
```
Comment (mrocklin) [inline docs/release-procedure.md]: wine -> twine

File: docs/release-procedure.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+````
+
+* Conda forge bots should pick up the change automatically. Just follow
```
Comment (mrocklin) [inline docs/release-procedure.md]: The Conda Forge bots should pick up the change automatically within an hour or two.  Then follow the instructions from the automatic e-mail that you will receive from Conda Forge.

drop trailing comma

**Comment by guillaumeeb** [comment]
Thanks @mrocklin, everything should be fixed.

File: docs/release-procedure.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+We may want to check their status while releasing.
+
+
+Release for dask-jobqueue, from within your fork:
+
+* Submit a PR that updates the release notes in `docs/source/changelog.rst`. 
+We submit a PR to inform other developers of the pending release, and possibly
+discuss its content.
+
+* Once the PR is merge, checkout master branch:
```
Comment (mrocklin) [inline docs/release-procedure.md]: merge -> merged.  Checkout master branch -> checkout the master branch

**Comment by mrocklin** [comment]
Thanks for writing this up @guillaumeeb !

**Pull request merged**